service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ReviewCMOControllerAboutToSubmitTest.java

```diff
@@ -0,0 +1,115 @@
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.fpl.controllers.cmo.ReviewCMOController;
+import uk.gov.hmcts.reform.fpl.enums.JudgeOrMagistrateTitle;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.HearingBooking;
+import uk.gov.hmcts.reform.fpl.model.ReviewDecision;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.model.common.JudgeAndLegalAdvisor;
+import uk.gov.hmcts.reform.fpl.model.order.CaseManagementOrder;
+import uk.gov.hmcts.reform.fpl.service.DocumentSealingService;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static uk.gov.hmcts.reform.fpl.enums.CMOReviewOutcome.JUDGE_REQUESTED_CHANGES;
+import static uk.gov.hmcts.reform.fpl.enums.CMOReviewOutcome.SEND_TO_ALL_PARTIES;
+import static uk.gov.hmcts.reform.fpl.enums.CMOStatus.RETURNED;
+import static uk.gov.hmcts.reform.fpl.enums.CMOStatus.SEND_TO_JUDGE;
+import static uk.gov.hmcts.reform.fpl.enums.HearingType.CASE_MANAGEMENT;
+import static uk.gov.hmcts.reform.fpl.model.order.CaseManagementOrder.sealFrom;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.element;
+import static uk.gov.hmcts.reform.fpl.utils.TestDataHelper.testDocumentReference;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(ReviewCMOController.class)
+@OverrideAutoConfiguration(enabled = true)
+public class ReviewCMOControllerAboutToSubmitTest extends AbstractControllerTest {
+
+    @MockBean
+    private DocumentSealingService documentSealingService;
+
+    protected ReviewCMOControllerAboutToSubmitTest() {
+        super("review-cmo");
+    }
+
+    @Test
+    void shouldSetReturnStatusAndRequestedChangesWhenJudgeRejectsOrder() {
+        CaseManagementOrder cmo = buildCMO();
+
+        CaseData caseData = CaseData.builder()
+            .draftUploadedCMOs(List.of(element(cmo)))
+            .reviewCMODecision(ReviewDecision.builder()
+                .changesRequestedByJudge("Please change XYZ")
+                .decision(JUDGE_REQUESTED_CHANGES).build())
+            .build();
+
+        CallbackRequest callbackRequest = CallbackRequest.builder().caseDetails(asCaseDetails(caseData)).build();
+
+        AboutToStartOrSubmitCallbackResponse response = postAboutToSubmitEvent(callbackRequest);
+
+        CaseData responseData = mapper.convertValue(response.getData(), CaseData.class);
```

Comment:
3 lines can be just replaced with:
CaseData responseData = extractCaseData(postAboutToSubmitEvent(caseData));